Ptt/Flows/BoardFlow/Controllers/SingleArticleViewController.swift

```diff
@@ -0,0 +1,359 @@
+                if !refreshControl.isRefreshing {
+                    activityIndicator.startAnimating()
+                }
+            }
+        } else {
+            activityIndicator.startAnimating()
+        }
+    }
+
+    @objc private func upvote() {
+    }
+
+    @objc private func reply() {
+    }
+
+    @objc private func share() {
+    }
+
+    @objc private func more() {
+    }
+}
+
+// MARK: - ASTableDataSource
+
+extension SingleArticleViewController: ASTableDataSource {
+
+    func tableNode(_ tableNode: ASTableNode, numberOfRowsInSection section: Int) -> Int {
+        return 2
+    }
+
+    func tableNode(_ tableNode: ASTableNode, nodeBlockForRowAt indexPath: IndexPath) -> ASCellNodeBlock {
+        let row = indexPath.row
+        guard let article = self.article else {
+            let nodeBlock: ASCellNodeBlock = {
+                return ASCellNode()
+            }
+            return nodeBlock
+        }
+        let nodeBlock: ASCellNodeBlock = {
+            let cell : ASCellNode
+            if row == 0 {
+                cell = ArticleMetaDataCellNode(article: article)
+                cell.backgroundColor = GlobalAppearance.backgroundColor
+            } else {
+                cell = ArticleContentCellNode(article: article)
+                cell.backgroundColor = PttColors.shark.color
+            }
+            return cell
+        }
+        return nodeBlock
+    }
+}
+
+// MARK: -
+
+private final class BoardNode: ASDisplayNode {
+
+    let tableNode = ASTableNode(style: .plain)
+    let activityIndicatorNode = ASDisplayNode { () -> UIView in
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.color = .lightGray
+        return activityIndicator
+    }
+    let bottomToolbarNode = BottomToolbarNode()
+
+    override init() {
+        super.init()
+        automaticallyManagesSubnodes = true
+
+        tableNode.backgroundColor = GlobalAppearance.backgroundColor
+    }
+
+    override func layoutSpecThatFits(_ constrainedSize: ASSizeRange) -> ASLayoutSpec {
+        let insets = UIEdgeInsets(top: .infinity,
+                                  left: 0,
+                                  bottom: 0,
+                                  right: 0)
+        let toolbarInsetSpec = ASInsetLayoutSpec(insets: insets, child: bottomToolbarNode)
+        let tableNodeSpec = ASOverlayLayoutSpec(child: tableNode, overlay: toolbarInsetSpec)
+        return ASOverlayLayoutSpec(child: tableNodeSpec, overlay: activityIndicatorNode)
+    }
+}
+
+private final class BottomToolbarNode: ASDisplayNode {
+
+    private let topBorderNode = ASDisplayNode()
+    let toolbarAreaNode = ArticleToolbarNode()
+    private let safeAreaNode = ASDisplayNode()
+
+    override init() {
+        super.init()
+
+        automaticallyManagesSubnodes = true
+        automaticallyRelayoutOnSafeAreaChanges = true
+
+        topBorderNode.backgroundColor = UIColor(red: 0.23, green: 0.23, blue: 0.23, alpha: 1.00)    // #3A3A3A
```

Comment:
Do you want to add this color to `ColorAssets` ?

Reply:
Great idea. Since I haven't finished implementing the toolbar yet, shall I do it for future PR?

Reply:
It can be done with another MR